fdbserver/IKeyValueStore.h

```diff
@@ -60,6 +78,11 @@ class IKeyValueStore : public IClosable {
 	                                                     int rowLimit = 1 << 30,
 	                                                     int byteLimit = 1 << 30) = 0;
 
+	// Range read implementation which utilizes a ResultWriter interface (defined above)
+	virtual Future<Void> readRange(KeyRangeRef keys, Reference<IReadRangeResultWriter> resultWriter, int rowLimit) {
```

Comment:
I think this having a default implementation is fine, since we don't plan to implement this for all storage engines. However, I would prefer to throw an error here (or just use `ASSERT(false)`. The user (storage server) is expected to first call `usesReadRangeResultWriter` before calling this method. If `usesReadRangeResultWriter` returns `false`, calling this method is illegal